### PR TITLE
Wrapper for Babel plugin transform runtime

### PR DESCRIPTION
**Pull request by snowystinger**
Need a wrapper for babel plugin transform runtime so we can transform to target esmodule

# ↪️ Pull Request

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/integration-tests/test/integration/babel-config-js-multitarget-transform-runtime/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  "presets": ["@parcel/babel-preset-env"],
+  "plugins": ["@parcel/transform-runtime"]
```
Comment (wbinnssmith) [inline packages/core/integration-tests/test/integration/babel-config-js-multitarget-transform-runtime/babel.config.js]: Maybe `@parcel/babel-plugin-transform-runtime` to be consistent with `@parcel/babel-preset-env`?

File: packages/utils/babel-plugin-transform-runtime/test/babel-plugin-transform-runtime.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      caller: {
+        name: 'foo',
+        version: '2.0.0',
+        targets: JSON.stringify({
+          esmodules: true,
+        }),
+      },
+    });
+
+    assert(!transformed.includes('class Foo'));
```
Comment (wbinnssmith) [inline packages/utils/babel-plugin-transform-runtime/test/babel-plugin-transform-runtime.test.js]: `assert` what it should include as well?

File: packages/utils/babel-plugin-transform-runtime/test/babel-plugin-transform-runtime.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  });
+
+  it('does not compile against targets passed through caller when the caller is not present', () => {
+    let {code: transformed} = babel.transformSync(input, {
+      configFile: false,
+      presets: [preset],
+      plugins: [plugin],
+    });
+
+    assert(!transformed.includes('class Foo'));
```
Comment (wbinnssmith) [inline packages/utils/babel-plugin-transform-runtime/test/babel-plugin-transform-runtime.test.js]: `assert` what it should include as well?

**Comment by snowystinger** [comment]
@wbinnssmith care to take another look?

File: packages/utils/babel-plugin-transform-runtime/src/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+module.exports = function parcelPluginTransformRuntime(api, opts, dirname) {
+  let name = api.caller(caller => caller && caller.name);
+  let version = api.caller(caller => caller && caller.version);
+
+  if (
+    name === 'parcel' &&
+    typeof version === 'string' &&
+    semver.satisfies(version, COMPATIBLE_PARCEL_BABEL_TRANSFORMER_SEMVER)
+  ) {
+    let targets = api.caller(caller => {
```
Comment (devongovett) [inline packages/utils/babel-plugin-transform-runtime/src/index.js:16]: We should remove this. Appears unused.

File: packages/utils/babel-plugin-transform-runtime/src/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      return caller && caller.targets;
+    });
+    let outputFormat = api.caller(caller => {
+      return caller && caller.env;
+    });
+    if (typeof targets !== 'string') {
```
Comment (devongovett) [inline packages/utils/babel-plugin-transform-runtime/src/index.js:22]: Maybe this meant to check `outputFormat` instead?

File: packages/utils/babel-plugin-transform-runtime/src/index.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      throw new Error('Expected targets to be a string');
+    }
+
+    return pluginTransformRuntime(
+      api,
+      {useESModules: outputFormat === 'esmodule', ...opts},
```
Comment (devongovett) [inline packages/utils/babel-plugin-transform-runtime/src/index.js:28]: We should merge `opts` first so we override `useESModules`

File: packages/transformers/babel/src/babel7.js [hunk 7]
```diff
@@ -45,6 +45,7 @@ export default async function babel7(
       name: 'parcel',
       version: transformerVersion,
       targets: JSON.stringify(babelOptions.targets),
+      env: asset.env.outputFormat,
```
Comment (devongovett) [inline packages/transformers/babel/src/babel7.js:48]: We should call this `outputFormat` since we aren't passing the entire `env` object

**Pull request merged**